src/actions.js

```diff
@@ -300,12 +300,18 @@ export function getTimeseriesEvents(uuid, start, end, params) {
     if (events && events.start === start && events.end === end) {
       return; // Up to date.
     } else if (!events || !events.isFetching) {
-      // Fetch it
-      dispatch(fetchTimeseriesEventsAction(uuid, start, end));
-
-      getTimeseries(uuid, start, end, params).then(results => {
-        receiveTimeseriesEvents(dispatch, uuid, start, end, results);
-      });
+      if (state.fakeData && state.fakeData[`timeseries-${uuid}`]) {
+        receiveTimeseriesEvents(dispatch, uuid, start, end, [
+          state.fakeData[`timeseries-${uuid}`]
+        ]);
+      } else {
+        // Fetch it
+        dispatch(fetchTimeseriesEventsAction(uuid, start, end));
+
+        getTimeseries(uuid, start, end, params).then(results => {
+          receiveTimeseriesEvents(dispatch, uuid, start, end, results);
+        });
+      }
```

Comment:
Ideally this functionality (lookup fakeData, if it isn't there do the request, return a Promise that will resolve to something to pass to receiveTimeseriesEvents) would be in its own async function in fakeData.js.

But in this case (high prio bug, and the other lookups are also still not that good) I can imagine we make this code a bit uglier still...